firmware/examples/TaskScheduler/source/main.cpp

```diff
@@ -0,0 +1,61 @@
+#include <cstdarg>
+
+#include "L0_LowLevel/uart0.hpp"
+#include "L2_Utilities/debug_print.hpp"
+#include "L4_Application/task.hpp"
+
+class PrinterTask : public rtos::Task<1024>
+{
+ public:
+    bool Setup() override
+    {
+        DEBUG_PRINT("Setup() for: %s", kName);
+        return true;
+    };
+    bool PreRun() override
+    {
+        DEBUG_PRINT("PreRun() for: %s", kName);
+        return true;
+    };
+    bool Run() override
+    {
+        run_count_ += 1;
+        DEBUG_PRINT("%s: %ld", message_, run_count_);
+        if (run_count_ == 10)
+        {
+            Delete();
+        }
+        return true;
+    };
+    PrinterTask(const char * task_name, const char * message)
```

Comment:
This class should follow the c++ class layout: https://google.github.io/styleguide/cppguide.html#Class_Format
Class constructor should be on top.